rspace-bench/src/main/scala/coop/rchain/rspace/bench/EvalBench.scala

```diff
@@ -0,0 +1,119 @@
+import coop.rchain.models.Par
+import coop.rchain.rholang.interpreter.{Env, Interpreter, Runtime}
+import org.openjdk.jmh.annotations._
+import coop.rchain.shared.PathOps.RichPath
+import monix.eval.Task
+import monix.execution.{ExecutionModel, Scheduler}
+import monix.execution.schedulers.{CanBlock, TrampolineScheduler}
+
+import scala.concurrent.Await
+import scala.concurrent.duration.Duration
+
+//for debug/check run from IDEA with
+//rspaceBench/jmh:run EvalBench.replayMvcEppST -i 1 -wi 0 -f 0 -t 1
+//on hyper-threaded machines for standalone check use -f <number of physical cores>
+//for example
+//java -jar target/scala-2.12/rspacebench_2.12-0.1.0-SNAPSHOT.jar EvalBench -i 10 -wi 5 -f 2 -t 2
+class EvalBench {
+  import EvalBench._
+
+  def processErrors(errors: Vector[Throwable]): Unit = if (errors.nonEmpty) {
+    throw new RuntimeException(
+      errors
+        .map(_.toString())
+        .mkString("Errors received during evaluation:\n", "\n", "\n"))
+  }
+
+  def prepareRunTask(state: MvcEppBenchState): Task[Vector[Throwable]] = {
+    val par = state.term.getOrElse(throw new Error("Failed to prepare executable rholand term"))
+    state.runtime.reducer
+      .inj(par)(state.rand)
+      .map(_ => state.runtime.readAndClearErrorVector())
+  }
+
+  //if we run multiple tests on a single-threaded scheduler
+  //they will compete on single execution queue
+  //therefore this test always limited to one thread
+  @Benchmark @Threads(1)
+  def replayMvcEppST(state: MvcEppBenchState): Unit = {
+    val runTask = prepareRunTask(state).executeOn(state.singleThreadedScheduler, forceAsync = false)
+    processErrors(
+      runTask.runSyncUnsafe(Duration.Inf)(state.singleThreadedScheduler, CanBlock.permit))
+  }
+
+  @Benchmark
+  def replayMvcEppMT(state: MvcEppBenchState): Unit = {
+    implicit val scheduler: Scheduler = monix.execution.Scheduler.Implicits.global
+    val runTask                       = prepareRunTask(state)
+    processErrors(Await.result(runTask.runAsync, Duration.Inf))
+  }
+}
+
+object EvalBench {
+
+  @State(Scope.Benchmark)
+  class MvcEppBenchState extends BenchStateBase {
+    val singleThreadedScheduler: Scheduler = TrampolineScheduler.apply(
+      Scheduler.singleThread(name = "mvcepp-1"),
+      ExecutionModel.SynchronousExecution)
+
+    override val rhoScriptSource: String = "/rholang/mvcepp.rho"
+  }
+
+  trait BenchStateBase {
+    private val dbDir: Path   = Files.createTempDirectory("rchain-storage-test-")
+    private val mapSize: Long = 1024 * 1024 * 1024
+
+    val rhoScriptSource: String
+    val runtime: Runtime       = Runtime.create(dbDir, mapSize)
+    val rand: Blake2b512Random = Blake2b512Random(128)
+    var term: Option[Par]      = None
+
+    /**
+      * until we add flag 'delete_lmdb_dir_on_close' for benchmarks and unit-tests
+      * this prevents periodic out of disk space failures
+      */
+    def deleteOldStorages(): Unit =
+      dbDir.getParent.toFile.listFiles
```

Comment:
This seems like a really fragile way to be getting our hands on `TMPDIR` and all of the test directories.

Is there a way for us to fix this problem in a general fashion so they we don't have to do this here?

Reply:
Well, I don't like this way too, but it looks almost perfect after removing these folders 10x times. I found that during benchmark runs all LMDB files are locked and can't be deleted, and if the are not locked - they never accessed anymore.

There is another way. We even discussed it approx one month ago. We have a lot of different storage related unit-tests and now with benchmarks count of "rchain-storage-test-" increases. That's not a problem on build server since it's cleaned before every build. I think we need to add a "test mode" flag to LMDBStore/LMDBTrieStore, and storage folder should delete automatically. Since this refactoring isn't related to benchmark and will affect numerous projects (Rspace, Rholang, Bench, etc). I think it should be extracted to a separate task. Removal of this function can be a final step of this task.

Reply:
Understood.  I'm fine with this approach here and now in this PR.  Just was curious to hear your thoughts.